### PR TITLE
Homework 1: Basic ls

### PR DESCRIPTION
**Pull request by psde**
Keeping it simple, hopefully.

### CODE REVIEW

File: course7/homeworks/mathias/ls/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+cmake_minimum_required(VERSION 3.4 FATAL_ERROR)
```
Comment (maciekgajewski) [inline course7/homeworks/mathias/ls/CMakeLists.txt:1]: Do you use any features introduced in in CMake 3.4?

File: course7/homeworks/mathias/ls/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+project(ls VERSION 1.0 LANGUAGES CXX)
+
+set(Boost_USE_STATIC_LIBS ON)
```
Comment (maciekgajewski) [inline course7/homeworks/mathias/ls/CMakeLists.txt:5]: why?

File: course7/homeworks/mathias/ls/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+find_package(Boost COMPONENTS filesystem REQUIRED)
+
+if(MSVC)
+    add_compile_options("/permissive-" "/W4" "/WX" "$<$<CONFIG:RELEASE>:/O2>")
+else()
+    add_compile_options("-std=c++14" "-Wall" "-Wextra" "-Werror" "$<$<CONFIG:RELEASE>:-O3>")
```
Comment (maciekgajewski) [inline course7/homeworks/mathias/ls/CMakeLists.txt:11]: Don't specify the build type directly in the CMakeLists.txt
Allow the person actually building the project to decided whether they want release or debug build, by providing CMAKE_BUILD_TYPE in command line and configure time.

**Review by maciekgajewski — APPROVED** [review]
Correct, but too complex!

File: course7/homeworks/mathias/ls/CMakeLists.txt (hunk 3 above)
Comment (psde) [reply course7/homeworks/mathias/ls/CMakeLists.txt:11]: This is a generator expression, it enables optimization on release builds only.

Comment (maciekgajewski) [reply course7/homeworks/mathias/ls/CMakeLists.txt:11]: Ah, OK
But I asked for for simple solution, and you couldn't resist adding all these complications :)

**Pull request merged**